src/Progress/Store.js

```diff
@@ -1,43 +1,81 @@
-import _ from 'lodash/fp'
 import { combineReducers } from 'redux'
 import BaseStore from '../Core/BaseStore'
 
 export const identifier = 'progress'
+
 const baseStore = BaseStore(identifier)
 
-const isVisible = d => d.code < 200 || d.code >= 400
-const getTotal = _.sumBy( () => 1 )
-const getDone = _.sumBy( e => e.erfuellt )
-const dashboardData = _.flow([
-    _.over([ getTotal, getDone ]),
-    ([ total, done ]) => ({ total, done })
-])
+function moduleIsVisible(module) { return module.code < 200 || module.code >= 400; }
+
+function groupBy(data, key) {
+
+    return data.reduce((storage, item) => {
+
+        var group = item[key] - 1;
+        storage[group] = storage[group] || [];
+        storage[group].push(item);
+        return storage;
+
+    }, []);
+}
+
+function getStudienfortschrittDisplayData(studienleistungen) {
+
+    studienleistungen = studienleistungen.meilensteine;
+    studienleistungen = groupBy(studienleistungen, 'fachsemester');
+
+    let studienfortschrittDisplayData = [];
+
+    studienleistungen.forEach(element => {
+
+        let object = {};
+        let zulassungsVoraussetzungFuerStudienleistung = element.find(d => d.code === element[0].fachsemester + 300);
+
+        object.label = element[0].fachsemester + '. Fachsemester';
+
+        object.prereq = !zulassungsVoraussetzungFuerStudienleistung || zulassungsVoraussetzungFuerStudienleistung.erfuellt;
+
+        object.completed = element.find(d => d.code === element[0].fachsemester + 200).erfuellt;
+
+        object.entries = element
+            .filter(moduleIsVisible)
+            .map(d => ({
+                ...d,
+                link: (d.format && `/exams/${d.format}s/${d.studiPruefungsId}`)
+            }));
```

Comment:
wenn ich das mal für mich übersetze, passiert in meinem kopf folgendes: 
ich hab keine ahnung, was `object` ist, genauso wenig, was `element` sein könnte. Wenn ich keine Ahnung habe, was in den variablen ist, dann kann das nur bedeuten, dass das eine äußerst allgemeine befehlssequenz ist, die auf sehr hohem Abstraktions-Niveau modelliert ist. 🤔
da dort "object" und "element" stehen, und da das technische javascript-begriffe sind, gehe ich davon aus, dass mit `module` ein ES6-module gemeint ist. 🧐
`d` ist auch völlig klar für mich: das muss auf jeden fall ein Objekt von sehr hohem Abstraktionsgrad sein, weil es ja in dem Kontext der abstrakten variablen/functions `object`, `element` und `moduleIsVisible` steht. und `d.format` macht auch sinn - wahrscheinlich haben alle allle alle objekte in levelup ein definiertes `format`. 🤓
aber moment mal - `d.studiPruefungsId` ?! oh nein... ich habe wahrscheinlich alles falsch verstanden.... 😫🤯

bitte bitte, aleks, mach weder mir noch kemal kopfschmerzen. umschreiben!